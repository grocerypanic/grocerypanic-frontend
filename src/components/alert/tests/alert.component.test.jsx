import { render, cleanup } from "@testing-library/react";
import React from "react";
import Assets from "../../../configuration/assets";
import Alert from "../alert.component";

describe("Setup Environment", () => {
  let utils;
  let message;

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterEach(cleanup);

  describe("when a message is not passed to the component", () => {
    beforeEach(() => {
      utils = render(<Alert />);
    });

    it("should render with with a non breaking space", () => {
      const node = utils.getByTestId("alert");
      expect(node.textContent).toBe(Assets.nonBreakingSpace);
    });

    it("should match the snapshot on file (styles)", () => {
      expect(utils.container.firstChild).toMatchSnapshot();
    });
  });

  describe("when a message is passed to the component", () => {
    beforeEach(() => {
      message = "Test Message";
      utils = render(<Alert message={message} />);
    });

    it("should render with the translated message", () => {
      expect(message).not.toBe(null);
      expect(utils.getByText(message)).toBeTruthy();
      expect(utils.getByTestId("alert")).toBeTruthy();
    });

    it("should match the snapshot on file (styles)", () => {
      expect(utils.container.firstChild).toMatchSnapshot();
    });
  });
});
