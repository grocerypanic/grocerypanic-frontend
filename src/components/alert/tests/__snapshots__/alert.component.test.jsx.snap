// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment when a message is not passed to the component should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .alert-light {
  padding-top: 5px;
  padding-bottom: 0px;
  margin-bottom: 0px;
}

<div
  class="c0"
>
  <div
    class="alert alert-light"
    data-testid="alert"
    role="alert"
  >
    Â 
  </div>
</div>
`;

exports[`Setup Environment when a message is passed to the component should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .alert-light {
  padding-top: 5px;
  padding-bottom: 0px;
  margin-bottom: 0px;
}

<div
  class="c0"
>
  <div
    class="alert alert-light"
    data-testid="alert"
    role="alert"
  >
    Test Message
  </div>
</div>
`;
